tee-worker/core-primitives/stf-executor/src/lib.rs

```diff
@@ -57,26 +57,39 @@ pub mod mocks;
 ///
 /// In case of success, it includes the operation hash, as well as
 /// any extrinsic callbacks (e.g. unshield extrinsics) that need to be executed on-chain
+///
+/// Litentry:
+/// we have made a few changes:
+/// - additionally for the success case, we add the encoded rpc response that will be
+///   passed back to the requester when the top status is `InSidechainBlock`
+/// - for failed top, we apply the parachain effects too
 #[derive(Clone, Debug, PartialEq)]
 pub enum ExecutionStatus {
-	Success(H256, Vec<OpaqueCall>),
-	Failure,
+	Success(H256, Vec<OpaqueCall>, Vec<u8>),
```

Comment:
Can we make this `Option<Vec<u8>>` because execution status of `TrustedOperation::indirect_call` shouldn't contain rpc response ?

Reply:
This is a valid concern, I'll leave it to another PR. In fact, I created a related issue earlier: https://github.com/litentry/litentry-parachain/issues/1922

It's about distinguishing II and DI in a broader sense

Reply:
ok 👍🏼